lib/openhab/dsl/rules/automation_rule.rb

```diff
@@ -87,21 +87,38 @@ def cleanup
         # @return [Queue] <description>
         #
         def create_queue(inputs)
-          case check_guards(event: inputs&.dig('event'))
+          case check_guards(event: extract_event(inputs))
           when true
             @run_queue.dup.grep_v(RuleConfig::Otherwise)
           when false
             @run_queue.dup.grep(RuleConfig::Otherwise)
           end
         end
 
+        #
+        # Extract the event object from inputs
+        # and merge other inputs keys/values into the event
+        #
+        # @param [Map] inputs rule inputs
+        #
+        # @return [Object] event object
+        #
+        def extract_event(inputs)
+          event = inputs&.dig('event')
+          unless event
+            event = Struct.new(:event, :attachment, :command).new
+            event.command = inputs&.dig('command')
+          end
+          add_attachment(event, inputs)
+        end
+
         #
         # Get the trigger_id for the trigger that caused the rule creation
         #
         # @return [Hash] Input hash potentially containing trigger id
         #
         def trigger_id(inputs)
-          inputs&.keys&.grep(/\.event$/)&.first&.chomp('.event')
```

Comment:
@boc-tothefuture I'd like to ask specifically about this. Was there a particular reason you chose to use `<triggerid>.event` to extract the trigger id instead of `module` below? I wasn't really sure whether there are cases where `module` doesn't exist perhaps? I tried running all the trigger tests locally, and they seem to pass using `module`, but I thought I should check with you first.

Reply:
My locale branch is on module.  I discovered that when I added support for attachments on cron triggers..

The only thing I can think of is that I either missed it or it wasn't there before 3.1.

Reply:
Thanks for the confirmation. I'll stick with module.